SimpleMQTTPublisher/Core/Utility.c

```diff
@@ -0,0 +1,23 @@
+#include <memory.h>
+#include "Utility.h"
+#include "ControlPacketFormat.h"
+
+int MQTT_WriteRemainingLength(unsigned char* pRemainingLengthBuffer, unsigned int dwBufferLength, unsigned int dwRestLength)
+{
+    MQTT_VariableByteInteger remainingLength = { 0 };
+    unsigned int dwRemainingLength = dwBufferLength - dwRestLength - 2;
+
+    if (MQTT_EncodeVariableByteInteger(dwRemainingLength, &remainingLength)) return -1;
+    if (dwRestLength < remainingLength.EncodedByteNum - 1) return -1;
+
+    if (pRemainingLengthBuffer != 0)
+    {
+        if (1 < remainingLength.EncodedByteNum)
+        {
+            memmove(pRemainingLengthBuffer + remainingLength.EncodedByteNum, pRemainingLengthBuffer + 1, dwRemainingLength);
+        }
+        memcpy(pRemainingLengthBuffer, remainingLength.EncodedByte, remainingLength.EncodedByteNum);
+    }
+
+    return remainingLength.EncodedByteNum - 1;
+}
```

Comment:
It is normal for `pRemainingLengthBuffer` to be called with `NULL (0)`.
The segmentation fault will not occur because the branch is made by the conditional expression of the if statement on line 13.

Reply:
@coderabbitai resolve